esphome/components/icnt86/icnt86.cpp

```diff
@@ -0,0 +1,97 @@
+#include "icnt86.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace icnt86 {
+
+static const char *const TAG = "icnt86";
+
+#define UBYTE uint8_t
+#define UWORD uint16_t
+#define UDOUBLE uint32_t
+
+void ICNT86Touchscreen::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up icnt86 Touchscreen...");
+
+  // Register interrupt pin
+  this->interrupt_pin_->pin_mode(gpio::FLAG_INPUT | gpio::FLAG_PULLUP);
+  this->interrupt_pin_->setup();
+  this->attach_interrupt_(interrupt_pin_, gpio::INTERRUPT_FALLING_EDGE);
+
+  // Perform reset if necessary
+  if (this->reset_pin_ != nullptr) {
+    this->reset_pin_->setup();
+    this->reset_();
+  }
+
+  this->x_raw_max_ = this->display_->get_native_width();
+  this->y_raw_max_ = this->display_->get_native_height();
+
+  this->conversion_to_resolution_ = false;
+  // Trigger initial read to activate the interrupt
+  this->store_.touched = true;
+}
+
+void ICNT86Touchscreen::update_touches() {
+  char buf[100];
+  char mask[1] = {0x00};
+  // Read report length
+  uint16_t data_len;
+
+  this->icnt_read_(0x1001, buf, 1);
+  uint8_t touch_count = buf[0];
+
+  if (touch_count == 0x00 || (touch_count > 5 || touch_count < 1)) {  // No new touch
+    return;
+  } else {
+    this->icnt_read_(0x1002, buf, touch_count * 7);
+    this->icnt_write_(0x1001, mask, 1);
+    ESP_LOGD(TAG, "Touch count: %d", touch_count);
+
+    for (UBYTE i = 0; i < touch_count; i++) {
```

Comment:
```suggestion
    for (uint8_t i = 0; i < touch_count; i++) {
```
please use uint8_t or uint16_t instead of UBYTE and UWORD etc or in this case you can use `auto` here.